src/sc2api/sc2_proto_to_pods.cc

```diff
@@ -202,9 +207,15 @@ bool Convert(const ObservationRawPtr& observation_raw, UnitPool& unit_pool, uint
         unit->is_on_screen = observation_unit.is_on_screen();
         unit->is_blip = observation_unit.is_blip();
 
-        unit->health = observation_unit.health();
+        const auto health = observation_unit.health();
+        if (health < unit->health)
+            unit_pool.AddUnitDamaged(unit);
```

Comment:
What about units which are under construction and under attack simultaneously? Maybe there is a trick to handle such cases too?

Reply:
good point. I don't know how the game handles health during construction. If you don't know I'll do some testing. If the current health of an un-attacked building is proportional to progress it would suggest possible approaches.

Reply:
I looked into this. It is a linear relationship, but it is weird. The first frame after a structure is created it has 10% max_health plus a small delta. The delta is different that the delta that is added to each subsequent frame. I agree we can skip for now, with a little more effort the exact relationship should be decipherable